RecoTracker/MkFitCore/src/KalmanUtilsMPlex.cc

```diff
@@ -0,0 +1,917 @@
+      c[10 * N + n] = a[6 * N + n] * b[1 * N + n] + a[7 * N + n] * b[4 * N + n] + a[8 * N + n] * b[7 * N + n];
+      c[11 * N + n] = 0;
+      c[12 * N + n] = a[10 * N + n] * b[0 * N + n] + a[11 * N + n] * b[3 * N + n] + a[12 * N + n] * b[6 * N + n];
+      c[13 * N + n] = a[10 * N + n] * b[1 * N + n] + a[11 * N + n] * b[4 * N + n] + a[12 * N + n] * b[7 * N + n];
+      c[14 * N + n] = 0;
+      c[15 * N + n] = a[15 * N + n] * b[0 * N + n] + a[16 * N + n] * b[3 * N + n] + a[17 * N + n] * b[6 * N + n];
+      c[16 * N + n] = a[15 * N + n] * b[1 * N + n] + a[16 * N + n] * b[4 * N + n] + a[17 * N + n] * b[7 * N + n];
+      c[17 * N + n] = 0;
+    }
+  }
+
+  void KalmanGain(const MPlexLS& A, const MPlex2S& B, MPlexL2& C) {
+    // C = A * B, C is 6x2, A is 6x6 sym , B is 2x2
+
+    typedef float T;
+    const idx_t N = NN;
+
+    const T* a = A.fArray;
+    ASSUME_ALIGNED(a, 64);
+    const T* b = B.fArray;
+    ASSUME_ALIGNED(b, 64);
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#include "KalmanGain62.ah"
+  }
+
+  inline void KHMult(const MPlexLH& A, const MPlexQF& B00, const MPlexQF& B01, MPlexLL& C) {
+    // C = A * B, C is 6x6, A is 6x3 , B is 3x6
+    KHMult_imp(A, B00, B01, C, 0, NN);
+  }
+
+  inline void KHC(const MPlexLL& A, const MPlexLS& B, MPlexLS& C) {
+    // C = A * B, C is 6x6, A is 6x6 , B is 6x6 sym
+
+    typedef float T;
+    const idx_t N = NN;
+
+    const T* a = A.fArray;
+    ASSUME_ALIGNED(a, 64);
+    const T* b = B.fArray;
+    ASSUME_ALIGNED(b, 64);
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#include "KHC.ah"
+  }
+
+  inline void KHC(const MPlexL2& A, const MPlexLS& B, MPlexLS& C) {
+    // C = A * B, C is 6x6 sym, A is 6x2 , B is 6x6 sym
+
+    typedef float T;
+    const idx_t N = NN;
+
+    const T* a = A.fArray;
+    ASSUME_ALIGNED(a, 64);
+    const T* b = B.fArray;
+    ASSUME_ALIGNED(b, 64);
+    T* c = C.fArray;
+    ASSUME_ALIGNED(c, 64);
+
+#include "K62HC.ah"
+  }
+
+  // //Warning: MultFull is not vectorized, use only for testing!
+  // template<typename T1, typename T2, typename T3>
+  // void MultFull(const T1& A, int nia, int nja, const T2& B, int nib, int njb, T3& C, int nic, int njc)
+  // {
+  // #ifdef DEBUG
+  //   assert(nja==nib);
+  //   assert(nia==nic);
+  //   assert(njb==njc);
+  // #endif
+  //   for (int n = 0; n < NN; ++n)
+  //     {
+  //       for (int i = 0; i < nia; ++i) {
+  // 	for (int j = 0; j < njb; ++j) {
+  // 	  C(n,i,j) = 0.;
+  // 	  for (int k = 0; k < nja; ++k) C(n,i,j) += A.ConstAt(n,i,k)*B.ConstAt(n,k,j);
+  // 	}
+  //       }
+  //     }
+  // }
+
+  // //Warning: MultTranspFull is not vectorized, use only for testing!
+  // // (careful about which one is transposed, I think rows and cols are swapped and the one that is transposed is A)
+  // template<typename T1, typename T2, typename T3>
+  // void MultTranspFull(const T1& A, int nia, int nja, const T2& B, int nib, int njb, T3& C, int nic, int njc)
```

Comment:
Addressed in Round 3 follow-up PR trackreco/cmssw#78 commit https://github.com/trackreco/cmssw/pull/78/commits/ed82f6fae7816eeb9d7f941e1b1a864f55027acd
the code was uncommented, in case it is useful